File: 13-none_nil_truthiness.rb
# Ruby 入面只有 nil 同 false 係 falsy，0、空字串同空 array 都係 truthy。
values = [nil, false, true, 0, 1, '', 'hello', [], [1]]

values.each do |value|
  if value
    puts "#{value.inspect} is truthy"
  else
    puts "#{value.inspect} is falsy"
  end
end

puts '---'

# 用 || 做 fallback 時，空字串喺 Ruby 仍然會保留。
def display_name(user)
  name = user[:name]

  name || 'Guest'
end

puts display_name({ name: 'Ada' })
puts display_name({ name: '' })
puts display_name({})

File: 13-none_nil_truthiness.py
# Python 入面 None、False、0、空字串同空 collection 都係 falsy。
values = [None, False, True, 0, 1, "", "hello", [], [1]]

for value in values:
    if value:
        print(f"{repr(value)} is truthy")
    else:
        print(f"{repr(value)} is falsy")

print("---")


# 用 or 做 fallback 時，空字串都會跌落 Guest。
def display_name(user):
    name = user.get("name")

    return name or "Guest"


print(display_name({"name": "Ada"}))
print(display_name({"name": ""}))
print(display_name({}))
